src/components/ProductMilestoneInterconnectionGraphPage/ProductMilestoneInterconnectionGraphPage.tsx

```diff
@@ -0,0 +1,207 @@
+
+import * as productMilestoneApi from 'services/productMilestoneApi';
+
+import { updateQueryParamsInURL } from 'utils/queryParamsHelper';
+
+const sharedArtifactsListColumns = ['identifier', 'product.name', 'productMilestone.version', 'build'];
+
+interface IProductMilestoneInterconnectionGraphPageProps {
+  componentId?: string;
+}
+
+export const ProductMilestoneInterconnectionGraphPage = ({
+  componentId = 'a1',
+}: IProductMilestoneInterconnectionGraphPageProps) => {
+  const { productMilestoneId } = useParams();
+
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  const { serviceContainerMilestone } = useServiceContainerMilestone();
+
+  const serviceContainerInterconnectionGraph = useServiceContainer(productMilestoneApi.getInterconnectionGraph);
+  const serviceContainerInterconnectionGraphRunner = serviceContainerInterconnectionGraph.run;
+
+  const serviceContainerSharedArtifacts = useServiceContainer(productMilestoneApi.getSharedArtifacts);
+  const serviceContainerSharedArtifactsRunner = serviceContainerSharedArtifacts.run;
+
+  const [limitNestingLevel, setLimitNestingLevel] = useState(false);
+  const [nestingLevel, setNestingLevel] = useState<number>(1);
+  const [searchValueProduct, setSearchValueProduct] = useState('');
+  const [searchValueMilestone, setSearchValueMilestone] = useState('');
+  const [showSharedArtifactsTable, setShowSharedArtifactsTable] = useState(false);
+
+  const graphDivRef = useRef<HTMLDivElement>(null);
+  const graphContentOffet = (graphDivRef.current?.children[1] as any)?.offsetTop;
+  const sharedArtifactsTableTopRef = useRef<HTMLDivElement>(null);
+
+  const { isFullscreen } = useFullscreen();
+
+  const loadSharedArtifacts = useCallback(
+    (milestone1: any, milestone2: any) => {
+      updateQueryParamsInURL(
+        { milestone1: milestone1 ? milestone1.id : '', milestone2: milestone2 ? milestone2.id : '', pageIndex: 1 },
+        componentId,
+        location,
+        navigate
+      );
+    },
+    [componentId, location, navigate]
+  );
+
+  const onSharedArtifactsListRender = useCallback(
+    () => sharedArtifactsTableTopRef.current?.scrollIntoView({ behavior: 'smooth' }),
+    []
+  );
+
+  useEffect(() => {
+    serviceContainerInterconnectionGraphRunner({
+      serviceData: { id: productMilestoneId },
+    });
+  }, [serviceContainerInterconnectionGraphRunner, productMilestoneId]);
+
+  useQueryParamsEffect(
+    ({ requestConfig } = {}) => {
+      serviceContainerSharedArtifactsRunner({ requestConfig });
+    },
+    {
+      componentId,
+      mandatoryQueryParams: { pagination: true, sorting: true, milestone: true },
+    }
+  );
+
+  useQueryParamsEffect(
+    ({ requestConfig } = {}) => {
+      if (requestConfig?.params.milestone1 && requestConfig?.params.milestone2) {
+        setShowSharedArtifactsTable(true);
+        sharedArtifactsTableTopRef.current?.scrollIntoView({ behavior: 'smooth' });
```

Comment:
Because `onSharedArtifactsListRender` is run just on component render. When edge has already been selected, and new edge is selected, that is use case when this specific scroll is needed.

Reply:
I meant to add javascript comment